pkg/lightning/backend/external/util_test.go

```diff
@@ -324,3 +325,210 @@ func TestKeyMinMax(t *testing.T) {
 	require.Equal(t, []byte("b"), BytesMax([]byte("a"), []byte("b")))
 	require.Equal(t, []byte("b"), BytesMax([]byte("b"), []byte("a")))
 }
+
+type EmbedA struct {
+	X string
+	Y int
+}
+
+type EmbedB EmbedA
+type EmbedC EmbedA
+type EmbedD EmbedA
+
+type InlineInternal struct {
+	I int    `json:"i"`
+	J string `json:"j"`
+}
+
+type EmbedExternal struct {
+	S int    `json:"s" external:"true"`
+	T string `json:"t" external:"true"`
+}
+
+type testStruct struct {
```

Comment:
with such a big struct, it's hard to tell what's tested, and what's not, can we split them into smaller cases

maybe you can use anonymous struct inside the test